src/chat-client/index.ts

```diff
@@ -726,24 +725,26 @@ class Chat extends ClientBase {
     const onLine = (line: string): void => {
       this._adminDebugLogger.info(`stdout from listener: ${line}`)
       try {
-        const messageObject = formatAPIObjectOutput(JSON.parse(line))
+        const messageObject = formatAPIObjectOutput(JSON.parse(line), null)
         if (messageObject.hasOwnProperty('error')) {
           throw new Error(messageObject.error)
         }
         if (messageObject.type !== 'chat' || !messageObject.msg) {
           return
         }
         const msgNotification: chat1.MsgNotification = messageObject
-        if (
-          // fire onMessage if it was from a different sender or at least a different device
-          // from this sender. Bots can filter out their own messages from other devices.
-          (options && options.showLocal) ||
-          (this.username &&
-            this.devicename &&
-            (msgNotification.msg.sender.username !== this.username.toLowerCase() ||
-              msgNotification.msg.sender.deviceName !== this.devicename))
-        ) {
-          onMessage(msgNotification.msg)
+        if (msgNotification.msg) {
+          if (
```

Comment:
Could merge these two `if`.

Reply:
I actually started that way -- it felt too hard to read for someone encountering it.